core/shared/utils/bh_memutils.c

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2024 Amazon Inc.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "bh_memutils.h"
+
+void *
+bh_memory_remap_slow(void *old_addr, size_t old_size, size_t new_size)
+{
+    void *new_memory =
+        os_mmap(NULL, new_size, MMAP_PROT_WRITE | MMAP_PROT_READ, 0, -1);
+    if (!new_memory) {
+        return NULL;
+    }
+    bh_memcpy_s(new_memory, new_size, old_addr, old_size);
```

Comment:
A potential issue is that bh_memcpy_s currently only supports size of uint32, so size may be truncated if it is larger than UINT32_MAX, e.g. in memory64 feature. How about using memcpy and add comment?